### PR TITLE
APP-1914: Fix tab notification not clearing

### PR DESCRIPTION
**Pull request by StylianosGakis**
This extracts the bottom nav badge logic into its own :notification-badge-data module, exposing only the two things that the rest of the app is interested about.
1. `TabNotificationBadgeService`
2. `CrossSellCardNotificationBadgeService`
One to interface with the badges in the bottom nav bar, and one for the "more insurances" section of the insurance tab.

Wrote a bunch extensive tests on it in `TabNotificationBadgeServiceTest` and `CrossSellCardNotificationBadgeServiceTest`.

This PR is on top of the extraction of apollo safe queries into the :apollo module PR since we needed to have access to them here.

Also currently to get the gql type Locale in the module I simply made the `getGraphqlLocaleFunctionQualifier` to pass a lambda which fetches the Locale since the LocaleManager is inside :app and can't be accessed there yet. My thought is that when the new language APIs PR is merged we can extract that information into a module where :notification-badge-data can directly depend on and avoid this extra step.

### CODE REVIEW

File: notification-badge-data/src/test/kotlin/com/hedvig/android/notification/badge/data/crosssell/card/CrossSellCardNotificationBadgeServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import com.hedvig.android.notification.badge.data.crosssell.FakeNotificationBadgeStorage
+import com.hedvig.android.notification.badge.data.crosssell.GetCrossSellsContractTypesUseCase
+import com.hedvig.android.notification.badge.data.storage.NotificationBadge
+import com.hedvig.android.notification.badge.data.storage.NotificationBadgeStorage
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+class CrossSellCardNotificationBadgeServiceTest {
```
Comment (hugokallstrom) [inline notification-badge-data/src/test/kotlin/com/hedvig/android/notification/badge/data/crosssell/card/CrossSellCardNotificationBadgeServiceTest.kt:17]: Really really good

File: notification-badge-data/src/test/kotlin/com/hedvig/android/notification/badge/data/crosssell/bottomnav/TabNotificationBadgeServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+import com.hedvig.android.notification.badge.data.storage.NotificationBadge
+import com.hedvig.android.notification.badge.data.storage.NotificationBadgeStorage
+import com.hedvig.android.notification.badge.data.tab.BottomNavTab
+import com.hedvig.android.notification.badge.data.tab.TabNotificationBadgeService
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+class TabNotificationBadgeServiceTest {
```
Comment (hugokallstrom) [inline notification-badge-data/src/test/kotlin/com/hedvig/android/notification/badge/data/crosssell/bottomnav/TabNotificationBadgeServiceTest.kt:23]: 👌

**Review by hugokallstrom — APPROVED** [review]
LGTM, GJ!

File: app/src/main/kotlin/com/hedvig/app/feature/loggedin/ui/LoggedInActivity.kt [hunk 3]
```diff
@@ -326,6 +328,7 @@ class LoggedInActivity : BaseActivity(R.layout.activity_logged_in) {
           .viewState
           .filterNotNull() // Emulate LiveData behavior of doing nothing until we get valid data
           .collectLatest { viewState: LoggedInViewState ->
+            d { "Stelios: viewState:$viewState" }
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/src/main/kotlin/com/hedvig/app/feature/loggedin/ui/LoggedInActivity.kt:331]: These always get me 😅 Removing!

**Pull request merged**